### PR TITLE
DCOS-16599: Introduce hooks to lint commit message

### PR DESCRIPTION
**Pull request by orlandohohmeier**
Add `conventional-changelog-lint` and a `post-commit` as well as adjust the `pre-push` hook to ensure that all commit messages are in line with the contributing guidelines. 

**Testing**
Please run `npm install` to install all dependencies as well as the new hooks and then do a well-typed and a _wrong_ test commit and try to push to a remote test(!) branch.

### CODE REVIEW

File: scripts/post-commit [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+title "Lint commit message..."
+
+linting_errors="$(npm run conventional-changelog-lint --silent -- -e; echo x$?)"
+
+if [  ${linting_errors##*x} -ne 0 ]
+then
+  warning "Wrong commit message format"
+  echo -e "Please adjust the commit message following the formatting rules" \
+    "as documented in our contributing guidelines.\n${linting_errors%x*}"
```
Comment (weblancaster) [inline scripts/post-commit]: We can even include the link https://github.com/dcos/dcos-ui/blob/master/CONTRIBUTING.md

Comment (orlandohohmeier) [inline scripts/post-commit]: Yeah, good point!

**Comment by weblancaster** [comment]
@orlandohohmeier I have made a test branch out of your branch, npm i and then made a small change.

Here what I was able to commit https://github.com/dcos/dcos-ui/commit/ef0e291146e762406fcab5c29e2659cdf676acf3 I'm guessing it's valid?

**Comment by MatApple** [comment]
@weblancaster you should `rm -rf node_modules` and try again.

**Comment by nLight** [comment]
It has two ✅  and an owner ✅  merge? :)

**Pull request merged**